### ISSUE TITLE
Improve unit test coverage

### ISSUE THREAD

**Issue by mAarnos** [mAarnos/Serkr#11]
Quite large parts of the prover are not unit tested. So adding more unit tests would be nice. However, some parts of the prover, such as the inference rules, some simplifcation rules, and the CNF translator appear to be not easily testable, since these tests would be extremely brittle and would break at the slightest change. Can something be done about this, or am I just plain wrong?

**Comment by Ticki** [comment]
Well, unit tests are entirely possible. Even the CNF translator is testable, by using "correctness" assertions (i.e. assert properties), instead of "equality" assertions (i.e. assert strict equality).

**Comment by mAarnos** [comment]
Yes, that seems like the best option. An added bonus is that it reduces the need for the internal parser, which I am planning on getting rid of.

**Comment by Ticki** [comment]
Interesting.
